packages/teleport/src/Account/ManageDevices/AddDevice/useAddDevice.ts

```diff
@@ -0,0 +1,75 @@
+import { useState, useEffect } from 'react';
+import useAttempt from 'shared/hooks/useAttemptNext';
+import Ctx from 'teleport/teleportContext';
+import cfg from 'teleport/config';
+
+export default function useAddDevice(
+  ctx: Ctx,
+  { token, fetchDevices, close }: Props
+) {
+  const [qrCode, setQrCode] = useState('');
+  const addDeviceAttempt = useAttempt('');
+  const fetchQrCodeAttempt = useAttempt('');
+
+  useEffect(() => {
+    fetchQrCodeAttempt.run(() =>
+      ctx.mfaService
+        .createRegisterChallenge(token, 'totp')
+        .then(res => setQrCode(res.totp.qrCode))
+    );
+  }, []);
+
+  function addTotpDevice(secondFactorToken: string, deviceName: string) {
+    addDeviceAttempt.setAttempt({ status: 'processing' });
+    ctx.mfaService
+      .addNewTotpDevice({
+        tokenId: token,
+        secondFactorToken,
+        deviceName,
+      })
+      .then(() => {
+        clearAttempt();
+        close();
+        fetchDevices();
```

Comment:
close should call fetch devices

Reply:
there wouldn't be a need to refresh the list if the user just changed their mind about adding a device and closed the dialog